### PR TITLE
Unique dashboard group, dashboard, and tab names in k8s testgrid

### PR DESCRIPTION
**Pull request by joshua-bone**
**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

File: config/prod/prow/jobs/custom/test-infra.yaml [hunk 1]
```diff
@@ -84,6 +84,7 @@ presubmits:
     run_if_changed: "^config/prod/prow/(jobs|k8s-testgrid)/.*.yaml"
     branches:
     - "master"
+    cluster: "knative-prow-build-cluster"
```
Comment (chaodaiG) [inline config/prod/prow/jobs/custom/test-infra.yaml:87]:
```suggestion
    cluster: "build-knative"
```

File: config/prod/prow/jobs/custom/test-infra.yaml [hunk 2]
```diff
@@ -539,6 +540,7 @@ postsubmits:
...
```
Comment (chaodaiG) [inline config/prod/prow/jobs/custom/test-infra.yaml:543]:
```suggestion
    cluster: "build-knative"
```

**Comment by joshua-bone** [comment]
Per @coryrc, the failing presubmit was caused by non-unique testgrid groups, dashboards, and dashboard tab names. Update PR to make all of those unique.

**Pull request merged**